### PR TITLE
DOCSP-12465 rust connection examples

### PR DESCRIPTION
**Pull request by ccho-mongodb**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-12465

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodborg-staging.corp.mongodb.com/506eafa/drivers/docsworker-xlarge/DOCSP-12465-rust-connection-examples/rust/#connect-to-mongodb-atlas

### CODE REVIEW

File: source/rust.txt [hunk 1]
```diff
@@ -37,35 +37,71 @@ See `Installation <https://github.com/mongodb/mongo-rust-driver#Installation>`__
...
 
-.. code-block:: rust
+.. tabs::
+
+   .. tab:: Sync
+      :tabid: rust-sync
+
+      .. code-block:: rust
+
+         // Be sure to set default-features=false and enable the "sync" feature in your Cargo.toml
```
Comment (nlarew) [inline source/rust.txt:52]: This comment feels important - is there a page we can link to that covers `Cargo.toml` for the Rust driver? (Maybe not since it's new docs for a new driver)

File: source/rust.txt [hunk 2]
```diff
@@ -37,35 +37,71 @@ See `Installation <https://github.com/mongodb/mongo-rust-driver#Installation>`__
...
+         use mongodb::{bson::doc, sync::Client};
+
+         fn main() -> mongodb::error::Result<()> {
+             // Get a handle to the deployment.
+             let client = Client::with_uri_str(
+                 "mongodb+srv://<username>:<password>@<cluster-url>/test?w=majority",
+             )?;
+
+             client
```
Comment (nlarew) [inline source/rust.txt:61]: Maybe add a comment explaining why we ping the admin db? Is this required to connect or just something we're doing for demo purposes?

File: source/rust.txt [hunk 3]
```diff
@@ -37,35 +37,71 @@ See `Installation <https://github.com/mongodb/mongo-rust-driver#Installation>`__
...
+   .. tab:: Sync
+      :tabid: rust-sync
+
+      .. code-block:: rust
+
+         // Be sure to set default-features=false and enable the "sync" feature in your Cargo.toml
+         use mongodb::{bson::doc, sync::Client};
+
+         fn main() -> mongodb::error::Result<()> {
+             // Get a handle to the deployment.
```
Comment (nlarew) [inline source/rust.txt:56]: nit: is "deployment" a term we're using commonly across the driver docs? I would expect this to be "cluster" instead.

**Review by nlarew — APPROVED** [review]
LGTM w/ some small comments

File: source/rust.txt (hunk 2 above)
Comment (ccho-mongodb) [reply source/rust.txt:61]: Not sure it's necessary, will ask Patrick.

File: source/rust.txt (hunk 3 above)
Comment (ccho-mongodb) [reply source/rust.txt:56]: I know choosing terms is on Tony Sansone's task list, but I agree that cluster makes more sense especially since it's the term used in the connection string. Will update all instances.

File: source/rust.txt (hunk 1 above)
Comment (ccho-mongodb) [reply source/rust.txt:52]: I think this is covered [here](https://github.com/mongodb/mongo-rust-driver#Installation), but maybe duplicating that or linking to that section would be useful in the tab. Will try something along those lines.

File: source/rust.txt (hunk 2 above)
Comment (patrickfreed) [reply source/rust.txt:64]: The `Client` will be connecting in the background, so the ping isn't necessary. Running a ping here is basically a test that the background connections are succeeding, but the same thing could be achieved via just the subsequent `list_database_names` call. Perhaps it makes sense to just remove the ping and associated `println` then if it adds confusion.

File: source/rust.txt [hunk 4]
```diff
@@ -37,35 +37,78 @@ See `Installation <https://github.com/mongodb/mongo-rust-driver#Installation>`__
...
+             println!("Connected successfully.");
+
+             // List the names of the databases in that cluster
+             for db_name in client.list_database_names(None, None)? {
+                 println!("{}", db_name);
+             }
+             Ok(())
+         }
+
+   .. tab:: Async
```
Comment (patrickfreed) [inline source/rust.txt:76]: Is there a possibility we could make the async tab the default one? We expect a large majority of users will be using the async API.

File: source/rust.txt [hunk 5]
```diff
@@ -37,35 +37,78 @@ See `Installation <https://github.com/mongodb/mongo-rust-driver#Installation>`__
...
+      :tabid: rust-async
+
+      Make sure you specified your async runtime. See
```
Comment (patrickfreed) [inline source/rust.txt:79]: We also think most users will be using the `tokio` runtime, which is the default and requires no configuration. Perhaps we could reword this to the something like the following:
> The default async runtime used by the driver is [`tokio`](tokio.rs). To use a different runtime, see [Configuring the async runtime](https://github.com/mongodb/mongo-rust-driver#configuring-the-async-runtime).

File: source/rust.txt (hunk 4 above)
Comment (ccho-mongodb) [reply source/rust.txt:76]: Sure, will update.

**Comment by ccho-mongodb** [comment]
Addressed comments, back to you @patrickfreed

**Comment by ccho-mongodb** [comment]
Addressed comments, back to you @patrickfreed !

**Review by patrickfreed — APPROVED** [review]
LGTM, thanks for updating this!
